src/track/keyutils.cpp

```diff
@@ -698,3 +698,60 @@ int KeyUtils::keyToCircleOfFifthsOrder(mixxx::track::io::key::ChromaticKey key,
         return s_sortKeysCircleOfFifthsLancelot[static_cast<int>(key)];
     }
 }
+
+// returns the pitch difference between two tracks, when they are played at the
+// same tempo. Assumes all inputs are valid.
+double KeyUtils::trackSyncPitchDifference(
+        mixxx::track::io::key::ChromaticKey key1,
+        mixxx::Bpm bpm1,
+        mixxx::track::io::key::ChromaticKey key2,
+        mixxx::Bpm bpm2) {
+    // convert minor keys to their relative majors
+    // we are interested in the notes used, not the mode
+    key1 = minorToRelativeMajor(key1);
```

Comment:
Did you consider this comment?

Reply:
Yes, as I understand, we need to add support to read cent values from the tracks right? I haven't been able to find the merged foundation PR. But I do agree it's important to handle this early. I guess we'd either have to accept cents as another parameter for each track, or we can pass in a "pitch" float value that will contain the major scale's root note as well as the cents. Is there something you have in mind for this?

Reply:
May idea was to pass the key as float and also make the key database coumn float. But we need to discuss.

Reply:
I also vote to accept pitch as a float in this function. 

The Key ID we are using right now also tells if the track is in Major / Minor modes. We didn't need that so we had to do the relative major conversion.

The float pitch value will be more convenient for this use case, because it does not depend on the mode, and it also encapsulates the detune. 

This float pitch can be derived from the Key and Cents column. Are there any other such derived columns in the Library?

Reply:
This is for sorting the Key column right? I feel sorting broadly based on the key wheel, and then within each key wheel value, sorting based on the cents, would be intuitive. So all tracks of a particular key are together, but the tracks that are more in tune, are higher within their key "group".